public/API/API_GetGameExtended.php

```diff
@@ -50,28 +50,97 @@
+        'ImageBoxArt' => $game->ImageBoxArt,
+        'Publisher' => $game->Publisher,
+        'Developer' => $game->Developer,
+        'Genre' => $game->Genre,
+        'Released' => $game->Released,
+        'IsFinal' => $game->IsFinal,
+        'RichPresencePatch' => md5($game->RichPresencePatch),
+        'GuideURL' => $game->GuideURL,
+        'Updated' => $game->Updated,
+    ];
 }
 
-if (empty($achData)) {
-    $gameData['Achievements'] = new ArrayObject(); // issue #484 - force serialization to {}
+/* Achievements Map
+    The maps here are return the data in the order that the v1 api expects, originally
+    the endpoint used function calls that made queries with the data in the
+    order that the endpoint returns but the Eloquent collections order the
+    data by how the columns are laid out in the database. This adds a few ms to the
+    response time, though not enough that it has any real impact.
+*/
+
+if (!$game->achievements->where('Flags', $flag)->isEmpty()) {
+    $gameAchievements = $game->achievements->where('Flags', $flag)->keyBy('ID')->map(function ($am) {
+        return [
+            'ID' => $am->ID,
+            'NumAwarded' => (int) Achievement::find($am->ID)->unlocks_total,
+            'NumAwardedHardcore' => (int) Achievement::find($am->ID)->unlocks_hardcore_total,
+            'Title' => $am->Title,
+            'Description' => $am->Description,
+            'Points' => $am->Points,
+            'TrueRatio' => $am->TrueRatio,
+            'Author' => $am->Author,
+            'DateModified' => Carbon::parse($am->DateModified)->format('Y-m-d H:i:s'),
+            'DateCreated' => Carbon::parse($am->DateCreated)->format('Y-m-d H:i:s'),
+            'BadgeName' => $am->BadgeName,
+            'DisplayOrder' => $am->DisplayOrder,
+            'MemAddr' => md5(Achievement::find($am->ID)->MemAddr),
+            'type' => $am->type,
+        ];
+    });
 } else {
-    foreach ($achData as &$achievement) {
-        $achievement['MemAddr'] = md5($achievement['MemAddr'] ?? null);
-    }
-    $gameData['Achievements'] = $achData;
+    $gameAchievements = new ArrayObject();
 }
-$gameData['Claims'] = getClaimData($gameID, false);
-$gameData['RichPresencePatch'] = md5($gameData['RichPresencePatch'] ?? null);
 
-$gameData['NumDistinctPlayersCasual'] = $gameData['NumDistinctPlayers'];
-$gameData['NumDistinctPlayersHardcore'] = $gameData['NumDistinctPlayers'];
+if (!$gameSetClaims) {
+    $gameClaims = [];
+} else {
+    $gameClaims = $gameSetClaims->map(function ($gc) {
+        return [
+            'User' => $gc->User,
+            'SetType' => $gc->SetType,
+            'GameID' => $gc->GameID,
+            'ClaimType' => $gc->ClaimType,
+            'Created' => Carbon::parse($gc->Created)->format('Y-m-d H:i:s'),
+            'Expiration' => Carbon::parse($gc->Finished)->format('Y-m-d H:i:s'),
+        ];
+    });
+}
 
-return response()->json($gameData);
+return response()->json(array_merge(
```

Comment:
nit: There's no need to instantiate 8 arrays here. `$gameData` is an array, and all the other fields can be in another array.

Reply:
Array fields compacted, return now only includes $gameData and a second array for the nested and individual stuff